server/graphql/core/src/loader/asset_location.rs

```diff
@@ -0,0 +1,85 @@
+use repository::asset_internal_location::{
+    AssetInternalLocationFilter, AssetInternalLocationRepository,
+};
+use repository::EqualFilter;
+use repository::{
+    location::{Location, LocationFilter, LocationRepository},
+    RepositoryError, StorageConnectionManager,
+};
+
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use std::collections::HashMap;
+
+pub struct AssetLocationLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+#[async_trait::async_trait]
+impl Loader<String> for AssetLocationLoader {
+    type Value = Vec<Location>;
+    type Error = RepositoryError;
+
+    async fn load(&self, ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let connection = self.connection_manager.connection()?;
+        let asset_location_repo = AssetInternalLocationRepository::new(&connection);
+        let location_repo = LocationRepository::new(&connection);
+
+        let locations =
+            asset_location_repo.query_by_filter(AssetInternalLocationFilter::new().asset_id(
+                EqualFilter::equal_any(ids.iter().map(String::clone).collect()),
+            ))?;
+
+        let mut asset_ids_by_location: HashMap<String, String> = HashMap::new();
+        for location in locations {
+            asset_ids_by_location.insert(location.location_id, location.asset_id);
+        }
+
+        let locations = location_repo.query_by_filter(LocationFilter::new().id(
+            EqualFilter::equal_any(asset_ids_by_location.clone().into_keys().collect()),
+        ))?;
+
+        let mut map: HashMap<String, Vec<Location>> = HashMap::new();
+        for location in locations {
+            let asset_id = asset_ids_by_location
+                .get(&location.location_row.id)
+                .unwrap_or(&"".to_string())
+                .to_owned();
+
+            let list = map
+                .entry(asset_id)
+                .or_insert_with(|| Vec::<Location>::new());
+            list.push(location);
+        }
+
+        Ok(map)
+    }
+}
+
+// #[cfg(test)]
+// mod tests {
```

Comment:
Couldn't get this test working? Would be good to have a test here for this if possible...